File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
def lcm(a, b):
    for i in range(max(a, b), (a*b)+1):
        if (i % a == 0) and (i % b == 0):
            return i


def solution(arr):
    first = 1
    for i in range(len(arr)):
        compare_num = lcm(first, arr[i])
        first = compare_num

    return compare_num

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java

class Solution {
    public static int lcm_2(int a, int b){
        int answer = 0;
        for(int i = Math.max(a,b); i <= a*b; i++){
            if(i % a == 0 && i % b == 0) {
                answer = i;break;
            }
        }
        return answer;
    }
    public int solution(int[] arr) {
        int answer = 1;
        int compare_num = 0;

        for(int i = 0; i < arr.length; i++){
            compare_num = lcm_2(answer,arr[i]);
            answer = compare_num;
        }

        return compare_num;
    }
}